### DISCUSSION TITLE
UTXO 与账户余额模型 · /utxo-account-models

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#520] [Announcements]
https://draveness.me/utxo-account-models

**Comment by sadikelong** [comment]
TOP2毕业的吧？？！！！

**Comment by draveness** [comment]
> @sadikelong
> TOP2毕业的吧？？！！！

不是

**Comment by hiberabyss** [comment]
首先指出一个 typo 哈, 下面这句话应该是输入之和大于输出之和吧?

> 所有的输入的 value 之和必须小于所有输出的 value 之和

想问下博主有没有研究过在存在多个 vin 的情况下怎么进行签名和验证?

**Comment by draveness** [comment]
> @hiberabyss
> 首先指出一个 typo 哈, 下面这句话应该是输入之和大于输出之和吧?
> 
> > 所有的输入的 value 之和必须小于所有输出的 value 之和
> 
> 想问下博主有没有研究过在存在多个 vin 的情况下怎么进行签名和验证?

研究过

**Comment by ytl123** [comment]
UTXO模型中得到余额不需要遍历整个区块链吧，遍历一下UTXO池就好了。

**Comment by fjchen7** [comment]
> @ytl123
> UTXO模型中得到余额不需要遍历整个区块链吧，遍历一下UTXO池就好了。

UTXO 池就是从区块中遍历得到的啊。

**Comment by mimirt** [comment]
謝謝你的文章，受益良多！

**Comment by chongyi** [comment]
想知道流程图用什么画的

**Comment by yummybian** [comment]
> 通过 txid 和 vout 两个字段，我们能够在区块链网络中定位到唯一一个 UTXO

txid就可以定位了吧，难道一个txid能对应到一个以上的UTXO？

**Comment by draveness** [comment]
@yummybian  Bitcoin 中就对应了多个输出

Regards,
Draven

**Comment by yummybian** [comment]
> @yummybian  Bitcoin 中就对应了多个输出 Regards, Draven
> […](#)
> On Mar 6, 2019, 4:48 PM +0800, xb ***@***.***>, wrote: > 通过 txid 和 vout 两个字段，我们能够在区块链网络中定位到唯一一个 UTXO > txid就可以定位了吧，难道一个txid能对应到一个以上的UTXO？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or mute the thread.

@draveness 你这样说也行。我想表达的是每个UTXO中txid字段值都是不一样的。

**Comment by OC0311** [comment]
有个小疑问 就是out 存储的是未花费的输出，那么已经花费的是怎么区分的啊

**Comment by draveness** [comment]
> 有个小疑问 就是out 存储的是未花费的输出，那么已经花费的是怎么区分的啊

已经花费的依靠之后花费这笔 out 的交易确定

**Comment by OC0311** [comment]
那如果一个utxo 是由两个不同地址的输入构成的 该如何表示啊   是在vin 中记录两条就是么？

**Comment by draveness** [comment]
> @jinjiangcc
> 那如果一个utxo 是由两个不同地址的输入构成的 该如何表示啊   是在vin 中记录两条就是么？

对

**Comment by elssm** [comment]
老哥是西安的嘛
